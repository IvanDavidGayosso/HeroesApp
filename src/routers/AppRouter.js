import React, { useContext } from "react";
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";
import { PrivateRoute } from "./PrivateRoute";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";
import { AuthContext } from "../auth/AuthContext"
import { PublicRoute } from "./PublicRoute";


export const AppRouter = () => {

    const { user } = useContext(AuthContext);

    return (
        <Router>
            <div>

                {/* A <Switch> looks through its children <Route>s and
                    renders the first one that matches the current URL. */}
                <Switch>

                    <PublicRoute 
                        path="/login" 
                        component = { LoginScreen } 
                        isAuthenticated = { user.logged }
                    />

                    <PrivateRoute 
                        path="/" 
                        component={ DashboardRoutes }
                        isAuthenticated = { user.logged }
                    />
                        
                </Switch>
            </div>
            </Router>
    )
}
