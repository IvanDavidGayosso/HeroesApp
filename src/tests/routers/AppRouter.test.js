import  { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Pruebas en <AppRouter />', () => {

    test('debe de mostrar el login si no está autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: false
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue } >
                <AppRouter/>
            </AuthContext.Provider>
            
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'debe de mostrar el componente marvel si está autenticado', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Ivan David'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue } >
                <AppRouter/>
            </AuthContext.Provider>
            
        );

        expect( wrapper.find('.navbar').exists() ).toBe( true );
    } );
    
})
