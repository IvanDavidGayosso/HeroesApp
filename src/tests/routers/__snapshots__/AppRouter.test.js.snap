// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no est√° autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
